### PR TITLE
Add ECDSA to Crypto package

### PR DESCRIPTION
**Pull request by yontyon**
In this PR we add a basic implementation for ECDSA in Crypto.
We use this infra to validate the operator requests in the default reconfiguration engine.

Note: further refactoring is needed to adjust the RSA infra to the new Signer and Verifier interfaces

### CODE REVIEW

**Comment by toly-kournik** [comment]
@yontyon, looks good.
I know that the current CryptographyWrapper location and namespace existed before, but probably it's time to move it to utils or a dedicated lib as it's a generic thing?

File: bftengine/src/bftengine/Crypto.hpp [hunk 1]
```diff
@@ -35,6 +35,40 @@ class CryptographyWrapper {
 
 // TODO(GG): define generic signer/verifier (not sure we want to use RSA)
 
+class Signer {
```
[the remark this answers is not in the document]
Comment (yontyon) [reply bftengine/src/bftengine/Crypto.hpp]: done

File: bftengine/src/bftengine/Crypto.hpp [hunk 2]
```diff
@@ -35,6 +35,40 @@ class CryptographyWrapper {
...
+  ECDSASigner(const std::string& path_to_pem_file);
+  ~ECDSASigner();
+  std::string sign(const std::string& data_to_sign) override;
+
+ private:
+  class Impl;
+  std::unique_ptr<Impl> impl_;
+};
+
+class ECDSAVerifier : public Verifier {
```
[the remark this answers is not in the document]
Comment (yontyon) [reply bftengine/src/bftengine/Crypto.hpp]: done

File: bftengine/include/bftengine/ReplicaConfig.hpp [hunk 3]
```diff
@@ -117,6 +117,8 @@ class ReplicaConfig : public concord::serialize::SerializableFactory<ReplicaConf
   CONFIG_PARAM(thresholdPublicKey_, std::string, "", "threshold crypto system bootstrap public key");
   std::vector<std::string> thresholdVerificationKeys_;
 
+  // Reconfiguration credentials
+  CONFIG_PARAM(pathToOperatprPublicKey_, std::string, "", "Path to the operator public key pem file");
```
Comment (toly-kournik) [inline bftengine/include/bftengine/ReplicaConfig.hpp:121]: typo

**Review by toly-kournik — APPROVED** [review]
LGTM

**Pull request merged**